[Yeh Yu-Cheng] iP

eltonyeh opened the pull request:
# Duke

> "Let our advance worrying become advance thinking and planning." - Winston Churchill [(Source)](https://www.inloox.com/company/blog/articles/the-62-most-inspiring-project-management-quotes/)

Duke frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ **SUPER FAST** to use

All you need to do is,
1. download it from [here](https://github.com/eltonyeh/ip/releases/tag/A-Jar-v2).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
    new Duke("tasks.txt").run();
}
```

Commit:
Add Gradle support

eltonyeh committed:
Add a skeletal version of Duke that echos the command and exit when type bye

eltonyeh committed:
Add the ability to list and store the commands

eltonyeh committed:
Add the ability of marking tasks as done

eltonyeh committed:
Add three more types of task

eltonyeh committed:
Add the error handling mechanism

eltonyeh committed:
Add ability to delete tasks

eltonyeh committed:
Add the input for the testing

eltonyeh committed:
Encapsulate all the messages to a class

eltonyeh committed:
Add new type of exceptions

eltonyeh committed:
Use enum to separate the task type

eltonyeh committed:
Add a TaskFile class to support file input

eltonyeh committed:
Add the functionality to transform tasks to file lines

eltonyeh committed:
Add new constructors for Tasks

eltonyeh committed:
Add the functionality to save tasks in txt file

eltonyeh committed:
Add the ability to recognize and query dates

eltonyeh committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/Task.java
#	src/main/java/ToDo.java

eltonyeh committed:
Merge branch 'branch-Level-8'

eltonyeh committed:
Solve some conflicts from the two branches

eltonyeh committed:
Make the code more OOP

eltonyeh committed:
Add all files into packages

eltonyeh committed:
Add JUnit test for the four Task classes

eltonyeh committed:
Fix the issue of not creating files occasionally

eltonyeh committed:
Add JavaDocs to all methods

eltonyeh committed:
Tweak the code to comply with the coding standard

eltonyeh committed:
Add the function of searching task for keyword

eltonyeh committed:
Merge branch 'branch-A-JavaDoc'

eltonyeh committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/task/Task.java

eltonyeh committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java

eltonyeh committed:
Reformat the code

eltonyeh committed:
Add classes for the commands

charltonator commented on src/main/java/duke/command/AddTaskCommand.java:
```diff
@@ -0,0 +1,39 @@
+import duke.Ui;
+import duke.task.Task;
+
+public class AddTaskCommand extends Command {
```
maybe it would be better if the class was named TaskAdder? AddTaskCommand sounds like maybe what a method would be called

charltonator commented on src/main/java/duke/command/ModifyTaskCommand.java:
```diff
@@ -0,0 +1,60 @@
+import duke.exception.IndexMismatchException;
+import duke.task.Task;
+
+public class ModifyTaskCommand extends Command {
```
Maybe can also change this class name to TaskModifier?

Zenlzb commented on src/test/java/duke/task/DeadlineTest.java:
```diff
@@ -0,0 +1,54 @@
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class DeadlineTest {
+    public static final LocalDate LOCALDATE = LocalDate.parse("2021-01-01");
```
Should this be LOCAL_DATE? or the next line be LOCALDATE_STRING?

Zenlzb commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,123 @@
+     */
+    public void refreshTask(TaskList taskList) {
+        try {
+            int numOfTask = taskList.size();
```
Should this be numOfTasks?

Zenlzb commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,170 @@
+     * @return The number of tasks.
+     */
+    public int size() {
+        return this.taskList.size();
```
I like that you wrote a header comment, but I believe it is not necessary for simple getter methods

Zenlzb commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,15 @@
+ */
+public class DukeException extends RuntimeException {
+    /**
+     * Constructors a {@code DukeException}.
```
Should this be "Constructs"?

Zenlzb commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    /**
+     * Return the time of the event.
```
Should this be "Returns"?

Zenlzb commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    /**
+     * Transforms the task to a single line that can be stored in a txt file.
```
Would "Returns a String" be a better way of describing this method? Considering nothing is happening to the existing task

Zenlzb reviewed (commented):
Not much to say, very meticulous

eltonyeh committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into branch-A-gradle

eltonyeh committed:
Automate project builds using Gradle

eltonyeh committed:
Merge branch 'branch-A-gradle'

eltonyeh committed:
Use checkStyle to detect coding style violations

eltonyeh committed:
Merge branch 'branch-A-CheckStyle'

eltonyeh committed:
Set up the GUI of duke

eltonyeh committed:
Finish the set up of the GUI

eltonyeh committed:
Merge branch 'branch-Level-10'

alyssa-savier commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,207 @@
+/**
+ * A class that deal with making sense of the user command.
+ */
+public class Parser {
```
I like your use of OOP.

alyssa-savier commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,68 @@
+     */
+    protected LocalDate timeStart;
+
+    /**
```
I like the inclusion of javadocs.

alyssa-savier commented on src/main/java/duke/task/ToDo.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    /**
+     * Transforms the task to a single line that can be stored in a txt file.
```
This description is a bit vague to me, perhaps you could explain what a line is?

alyssa-savier commented on src/test/java/duke/task/ToDoTest.java:
```diff
@@ -0,0 +1,37 @@
+
+public class ToDoTest {
+    @Test
+    public void constructorWithEmptyDescriptionTest() {
```
Could this method name be shortened?

alyssa-savier commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,70 @@
+     * @param tasks The list to which a new task is added.
+     * @param task  The task added to the list.
+     */
+    public static String addTaskMessage(TaskList tasks, Task task) {
```
Does this follow the java coding naming standard?

alyssa-savier commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,125 @@
+     * @return A {@code TaskList} based on the file content.
+     * @throws DukeException If anything wrong happens.
+     */
+    public TaskList parseToTaskList() throws DukeException {
```
Would it be better if this was named parseTaskList()?

eltonyeh committed:
Add assertion statements in the code

We can not ensure some desired properties holds in our code.

Let's use Java assertion to mark and check some important properties/
assumptions that should hold in various places of the code.

Java assertion can ensure these assumptions hold without modifying the
code structure.

eltonyeh committed:
Improve the code quality

The code structure may confuse other code readers.

Let's restructure the code to make it more understandable.

eltonyeh committed:
Merge pull request #2 from eltonyeh/branch-A-Assertions

Add assertion statements in the code

eltonyeh committed:
Merge branch 'master' of https://github.com/eltonyeh/ip into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

eltonyeh committed:
Merge pull request #1 from eltonyeh/branch-A-CodeQuality

Improve the code quality

eltonyeh committed:
Set up continuous integrations by using GitHub actions

eltonyeh committed:
Set up the ability to update tasks

eltonyeh committed:
Merge branch 'branch-C-Update'

eltonyeh committed:
Add some javadoc comments

eltonyeh committed:
Change the display format of invalid input in the GUI

eltonyeh committed:
Modify the padding of the GUI

eltonyeh committed:
Add the ability to undone tasks

eltonyeh committed:
Add a profile picture and an user guide

eltonyeh committed:
Set theme jekyll-theme-cayman

eltonyeh committed:
Style the GUI to make it more attractive

eltonyeh committed:
Merge branch 'master' of https://github.com/eltonyeh/ip

eltonyeh committed:
Change the JavaFX API version of FXML file from 16 to 11

eltonyeh committed:
Fix the unclickable textfield issue

eltonyeh committed:
Update the UI screenshot

eltonyeh committed:
Add some test for `TaskList`

eltonyeh committed:
Fix some Javadoc style violations

eltonyeh committed:
Update README.md
